XRTK.Oculus/Packages/com.xrtk.oculus/Editor/OculusManifestPreprocessor.cs

```diff
@@ -25,60 +25,81 @@
-			}
-
-			System.IO.File.WriteAllText(dstFile, manifestText);
-			AssetDatabase.Refresh();
-		}
-
-		[MenuItem("Mixed Reality Toolkit/Tools/Remove AndroidManifest.xml", false, 100001)]
-		public static void RemoveAndroidManifest()
-		{
-			AssetDatabase.DeleteAsset("Assets/Plugins/Android/AndroidManifest.xml");
-			AssetDatabase.Refresh();
-		}
-	}
+    public class OculusManifestPreprocessor
+    {
+        [MenuItem("Mixed Reality Toolkit/Tools/Create Oculus Quest compatible AndroidManifest.xml", false, 100000)]
+        public static void GenerateManifestForSubmission()
+        {
+            var so = ScriptableObject.CreateInstance(typeof(OculusUpdaterStub));
+            var script = MonoScript.FromScriptableObject(so);
+            string assetPath = AssetDatabase.GetAssetPath(script);
+            string editorDir = Directory.GetParent(assetPath).FullName;
+            string srcFile = editorDir + "/AndroidManifest.OVRSubmission.xml";
+
+            if (!File.Exists(srcFile))
+            {
+                Debug.LogError("Cannot find Android manifest template for submission." +
+                    " Please delete the OVR folder and reimport the Oculus Utilities.");
+                return;
+            }
+
+            string manifestFolder = Application.dataPath + "/Plugins/Android";
+
+            if (!Directory.Exists(manifestFolder))
+                Directory.CreateDirectory(manifestFolder);
+
+            string dstFile = manifestFolder + "/AndroidManifest.xml";
+
+            if (File.Exists(dstFile))
+            {
+                Debug.LogWarning("Cannot create Oculus store-compatible manifest due to conflicting file: \""
+                    + dstFile + "\". Please remove it and try again.");
+                return;
+            }
+
+            string manifestText = File.ReadAllText(srcFile);
+            int dofTextIndex = manifestText.IndexOf("<!-- Request the headset DoF mode -->");
+            if (dofTextIndex != -1)
+            {
+                //Forces Quest configuration.  Needs flip for Go/Gear viewer
+                string headTrackingFeatureText = "<uses-feature android:name=\"android.hardware.vr.headtracking\" android:version=\"1\" android:required=\"true\" />";
+                manifestText = manifestText.Insert(dofTextIndex, headTrackingFeatureText);
+            }
+            else
+            {
+                Debug.LogWarning("Manifest error: unable to locate headset DoF mode");
+            }
+
+            int handTrackingTextIndex = manifestText.IndexOf("<!-- Request the headset handtracking mode -->");
+            if (handTrackingTextIndex != -1)
+            {
+
+                    bool handTrackingEntryNeeded = true; // (targetHandTrackingSupport != OVRProjectConfig.HandTrackingSupport.ControllersOnly);
+                    bool handTrackingRequired = false; // (targetHandTrackingSupport == OVRProjectConfig.HandTrackingSupport.HandsOnly);
+                    if (handTrackingEntryNeeded)
```

Comment:
handTrackingEntryNeeded is always true?

Reply:
For now until we have a Platform system with Hands configuration to alter it
It's set to enable projects to be able to work with EITHER hands or controllers